GeneratorInterface/RivetInterface/src/HiggsTemplateCrossSections.cc

```diff
@@ -0,0 +1,549 @@
+#ifndef TRUTHRIVETTOOLS_HIGGSTEMPLATECROSSSECTIONS_CC
+#define TRUTHRIVETTOOLS_HIGGSTEMPLATECROSSSECTIONS_CC
+
+// -*- C++ -*-
+#include "Rivet/Analysis.hh"
+#include "Rivet/Particle.hh"
+#include "Rivet/Projections/FastJets.hh"
+
+// Definition of the StatusCode and Category enums
+//#include "HiggsTemplateCrossSections.h"
+#include "SimDataFormats/HTXS/interface/HiggsTemplateCrossSections.h"
+
+namespace Rivet {
+  
+  /// @class HiggsTemplateCrossSections 
+  /// @brief  Rivet routine for classifying MC events according to the Higgs template cross section categories
+  /// @author Jim Lacey (DESY) <james.lacey@cern.ch,jlacey@desy.de>
+  /// @author Dag Gillberg (Carleton University) <dag.gillberg@cern.ch>
+  class HiggsTemplateCrossSections : public Analysis {
+  public:
+    // Constructor
+    HiggsTemplateCrossSections()
+      : Analysis("HiggsTemplateCrossSections"),
+	m_HiggsProdMode(HTXS::UNKNOWN) {}
+
+  public:
+
+    /// @name Utility methods
+    /// Methods to identify the Higgs boson and
+    /// associated vector boson and to build jets
+    /// @{
+
+    /// follow a "propagating" particle and return its last instance
+    Particle getLastInstance(Particle ptcl) {
+      if ( ptcl.genParticle()->end_vertex() ) {
+	if ( !hasChild(ptcl.genParticle(),ptcl.pdgId()) ) return ptcl;
+	else return getLastInstance(ptcl.children()[0]);
+      }
+      return ptcl;
+    }
+    
+    /// @brief Whether particle p originate from any of the ptcls
+    bool originateFrom(const Particle& p, const Particles& ptcls ) {
+      const GenVertex* prodVtx = p.genParticle()->production_vertex();
+      if (prodVtx == nullptr) return false;
+      // for each ancestor, check if it matches any of the input particles
+      for (auto ancestor:particles(prodVtx, HepMC::ancestors)){ 
```

Comment:
can this and the one below be (const) & instead?

Reply:
something like
`(const) &ancestor:particles`
instead of
`auto ancestor:particles`
?

Reply:
I get the error
```
HiggsTemplateCrossSections.cc:47:13: error: ISO C++ forbids declaration of 'type name' with no type [-fpermissive]
       for ((const) auto &ancestor:particles(prodVtx, HepMC::ancestors)){
             ^~~~~
```

Reply:
const auto & ancestor:particles